### PR TITLE
Add linkedlist

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This adds an implementation for a linked list.

@neithernut please review. @manuelmessner and @TheNeikos, you are welcome, too!

### CODE REVIEW

File: src/libreset/ll.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    struct ll_element*  next;
+    void*               data;
+};
+
+struct ll {
+    struct ll_element* head;
+};
+
+struct ll*
+ll_alloc(void);
```
Comment (neithernut) [inline src/libreset/ll.h]: We won't need an allocator. Just put a `struct ll` into the node struct.

File: src/libreset/ll.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    for (iter = ll->head; iter; iter = iter->next) {
+        if (iter == del) {
+            last->next = iter->next;
+            free(iter);
+            break;
+        }
+
+        last = iter;
+    }
```
Comment (neithernut) [inline src/libreset/ll.c:65]: This does not work if the first element is deleted.

File: src/libreset/ll.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    struct ll* ll,
+    struct ll_element* del
+) {
+    struct ll_element* iter;
+    struct ll_element* last;
+
+    for (iter = ll->head; iter; iter = iter->next) {
+        if (iter == del) {
+            if (iter == ll->head) {
+                ll->head = iter->next;
```
Comment (neithernut) [inline src/libreset/ll.c]: You check every element for being the head element?
Why don't you just check whether you delete the first element before running into the loop?

Comment (matthiasbeyer) [inline src/libreset/ll.c]: That was indeed stupid.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**